include/libopencm3/stm32/common/usart_common_fl_0.h

```diff
@@ -0,0 +1,302 @@
+#define USART_CR2_ABRMOD_MASK		0x3
+#define USART_CR2_ABRMOD_STARTBIT	(0 << USART_CR2_ABRMOD_SHIFT)
+#define USART_CR2_ABRMOD_FALLTOFALL	(1 << USART_CR2_ABRMOD_SHIFT)
+
+#define USART_CR2_ABREN			(1 << 20)
+#define USART_CR2_MSBFIRST		(1 << 19)
+#define USART_CR2_DATAINV		(1 << 18)
+#define USART_CR2_TXINV			(1 << 17)
+#define USART_CR2_RXINV			(1 << 16)
+#define USART_CR2_SWAP			(1 << 15)
+#define USART_CR2_LINEN			(1 << 14)
+
+#define USART_CR2_STOP_SHIFT		12
+#define USART_CR2_STOP			(3 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_1_0BIT		(0 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_2_0BIT		(2 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_1_5BIT		(3 << USART_CR2_STOP_SHIFT)
+
+#define USART_CR2_CLKEN			(1 << 11)
+#define USART_CR2_CPOL			(1 << 10)
+#define USART_CR2_CPHA			(1 << 9)
+#define USART_CR2_LBCL			(1 << 8)
+#define USART_CR2_LBIDE			(1 << 6)
+#define USART_CR2_LBDL			(1 << 5)
+#define USART_CR2_ADDM			(1 << 4) /* Obsolete, use ADDM7 */
+#define USART_CR2_ADDM7			(1 << 4)
+
+/* USART_CR3 Values ---------------------------------------------------------*/
+
+#define USART_CR3_WUFIE			(1 << 22)
+
+#define USART_CR3_WUS_SHIFT		20
+#define USART_CR3_WUS			(3 << USART_CR3_WUS_SHIFT)
+#define USART_CR3_WUS_ADDRMATCH		(0 << USART_CR3_WUS_SHIFT)
+#define USART_CR3_WUS_STARTBIT		(2 << USART_CR3_WUS_SHIFT)
+#define USART_CR3_WUS_RXNE		(3 << USART_CR3_WUS_SHIFT)
+
+#define USART_CR3_SCARCNT_SHIFT		17
+#define USART_CR3_SCARCNT		(7 << USART_CR3_SCARCNT_SHIFT)
+#define USART_CR3_SCARCNT_DISABLE	(0 << USART_CR3_SCARCNT_SHIFT)
+#define USART_CR3_SCARCNT_VAL(x)	((x) << USART_CR3_SCARCNT_SHIFT)
+
+#define USART_CR3_DEP			(1 << 15)
+#define USART_CR3_DEM			(1 << 14)
+#define USART_CR3_DDRE			(1 << 13)
+#define USART_CR3_OVRDIS		(1 << 12)
+#define USART_CR3_ONEBIT		(1 << 11)
+#define USART_CR3_CTSIE			(1 << 10)
+#define USART_CR3_CTSE			(1 << 9)
+#define USART_CR3_RTSE			(1 << 8)
+#define USART_CR3_DMAT			(1 << 7)
+#define USART_CR3_DMAR			(1 << 6)
+#define USART_CR3_SCEN			(1 << 5)
+#define USART_CR3_NACK			(1 << 4)
+#define USART_CR3_HDSEL			(1 << 3)
+#define USART_CR3_IRLP			(1 << 2)
+#define USART_CR3_IREN			(1 << 1)
+#define USART_CR3_EIE			(1 << 0)
+
+/* USART_GTPR Values --------------------------------------------------------*/
+
+#define USART_GTPR_GT_SHIFT		8
+#define USART_GTPR_GT			(0xFF << USART_GTPR_GT_SHIFT)
+#define USART_GTPR_GT_VAL(x)		((x) << USART_GTPR_GT_SHIFT)
+
+#define USART_GTPR_PSC_SHIFT		0
+#define USART_GTPR_PSC			(0xFF << USART_GTPR_PSC_SHIFT)
+#define USART_GTPR_PSC_VAL(x)		((x) << USART_GTPR_PSC_SHIFT)
+
+
+/* USART_RTOR Values --------------------------------------------------------*/
+
+#define USART_RTOR_BLEN_SHIFT		24
+#define USART_RTOR_BLEN			(0xFF << USART_RTOR_BLEN_SHIFT)
+#define USART_RTOR_BLEN_VAL(x)		((x) << USART_RTOR_BLEN_SHIFT)
+
+#define USART_RTOR_RTO_SHIFT		0
+#define USART_RTOR_RTO			(0xFF << USART_RTOR_RTO_SHIFT)
+#define USART_RTOR_RTO_VAL(x)		((x) << USART_RTOR_RTO_SHIFT)
```

Comment:
They were not actually used so I've converted them to the more common 'shift/mask' variant.